src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -1630,4 +1686,17 @@ private void logAttackMoves(final List<ProTerritory> prioritizedTerritories) {
       }
     }
   }
+
+  private boolean canAirSafelyLandAfterAttack(final Unit unit, final Territory t) {
+    final boolean isAdjacentToAlliedFactory =
+        Matches.territoryHasNeighborMatching(data,
+            ProMatches.territoryHasInfraFactoryAndIsAlliedLand(player, data)).match(t);
+    final int range = TripleAUnit.get(unit).getMovementLeft();
+    final int distance =
+        data.getMap().getDistance_IgnoreEndForCondition(ProData.unitTerritoryMap.get(unit), t,
+            ProMatches.territoryCanMoveAirUnitsAndNoAA(player, data, true));
+    final boolean usesMoreThanHalfOfRange = distance > range / 2;
+    return isAdjacentToAlliedFactory || !usesMoreThanHalfOfRange;
```

Comment:
This return statement indicates there is a lot of unnecessary execution, and a SRP violation. We know very early if `isAdjacentToAlliedFactory` is true or not.  I recommend making `canAirSafelyLandAfterAttack` dead simple, and using methods rather than variable names to encapsulate the logic, ie:
`return (canAirSafelyLandAfterAttack() || !usesMoreThanHalfOfRange())`

Then in turn the two helper methods are more simple (KISS principle)